src/org/rascalmpl/library/util/PathConfig.java

```diff
@@ -438,180 +401,253 @@ private static ISourceLocation inferProjectRoot(ISourceLocation member) {
         return current;
     }
 	/**
-	 * This will create a PathConfig by learning from the MANIFEST/RASCAL.MF file where the sources
-     * are, which libraries to reference and which classpath entries to add. If this PathConfig is
-     * for the interpreter it adds more folders to the source path than if its for the compiler.
+	 * This creates a PathConfig instance by learning from the `MANIFEST/RASCAL.MF` file where the sources
+     * are, and from `pom.xml` which libraries to reference. If this PathConfig is
+     * for the interpreter it adds more folders to the source path than if its for the compiler. 
      * 
-     * If library dependencies exist for open projects in the same IDE, via the lib://libName, project://libName
-     * correspondence, then the target and source folders of the projects are added rather then
-     * the jar files. For compiler configs this works differently than for interpreter configs.
-     * The latter adds source folders to the sources while the former adds target folders to the libraries.
+     * In the future
+     * we'd like the source folders also configured by the pom.xml but for now we read it from RASCAL.MF for
+     * the sake of efficiency (reading pom.xml requires an XML parse and DOM traversal.) Also we need
```

Comment:
note, there is a [specific maven library that is just for reading the pom.xml](https://maven.apache.org/archives/maven-1.x/reference/maven-model/3.0.2/apidocs/org/apache/maven/project/io/xpp3/MavenXpp3Reader.html)

Reply:
that's nice but it's not enough. we need recursive dependency resolution. So what @rodinaarssen is doing is more useful.

Reply:
sure, but this was about reading the source folder from the pom, just saying, that's possible without custom code on our side.